staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -1048,77 +1049,86 @@ func seedMultiLevelData(ctx context.Context, store storage.Interface) (string, [
 }
 
 func RunTestGetListNonRecursive(ctx context.Context, t *testing.T, store storage.Interface) {
-	prevKey, prevStoredObj := TestPropagateStore(ctx, t, store, &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "prev"}})
-
+	key, prevStoredObj := testPropagateStore(ctx, t, store, &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo"}})
```

Comment:
I broke this one in [#113427](https://github.com/kubernetes/kubernetes/pull/113427)

Previously both items (prevStoredObj and storeObj) had the same key - the above PR broke it.
I'm fixing it in this PR.

Reply:
We should either be extremely explicit about the key, or always allow it to be generated. Not sure we gain a lot other than mental overhead by mixing the two.

Reply:
I don't understand your comment.

We're relying on the object namespace/name to be the key, so we're not mixing anything now.
Simply the rule is - the same object (defined by (namespace,name)) - same key, different object - different keys.

The rule is fairly explicit after the change from the previous PR.

Reply:
Yes, I think now we're consistent as long as we're not specifying key but instead letting it be computed from the object being passed in.